src/olm/session/mod.rs

```diff
@@ -207,17 +209,25 @@ impl Session {
         if self.has_received_message() {
             OlmMessage::Normal(message)
         } else {
-            let message = PreKeyMessage::new(
-                self.session_keys.one_time_key,
-                self.session_keys.base_key,
-                self.session_keys.identity_key,
-                message,
-            );
+            let message = PreKeyMessage::new(self.session_keys, message);
 
             OlmMessage::PreKey(message)
         }
     }
 
+    /// Get the keys associated with this session.
+    pub fn session_keys(&self) -> SessionKeys {
+        self.session_keys
+    }
+
+    #[cfg(feature = "low-level-api")]
+    /// Get the [`MessageKey`] to encrypt the next message.
+    ///
+    /// **Note**: This *must* be used to encrypt the next message.
```

Comment:
This isn't strictly true, a `Session` supports out of order decryption, you just need to make sure you're not too much out of order. Though not sure if we want to signal that to the user.

Reply:
What about "This must be used to encrypt a message"?